vra/resource_deployment.go

```diff
@@ -370,17 +458,218 @@ func resourceDeploymentUpdate(d *schema.ResourceData, m interface{}) error {
+		blueprintId, catalogItemId := "", ""
+		if v, ok := d.GetOk("catalog_item_id"); ok {
+			catalogItemId = v.(string)
+		}
+
+		if v, ok := d.GetOk("blueprint_id"); ok {
+			blueprintId = v.(string)
+		}
+
+		// If catalog_item_id is provided, get the catalog item schema deployment with the catalog item
+		if catalogItemId != "" {
+			catalogItemVersion := ""
+			if v, ok := d.GetOk("catalog_item_version"); ok {
+				catalogItemVersion = v.(string)
+			}
+
+			if v, ok := d.GetOk("inputs"); ok {
+				inputsSchemaMap, err := getCatalogItemSchema(apiClient, catalogItemId, catalogItemVersion)
+				if err != nil {
+					return err
+				}
+
+				log.Printf("Inputs Schema: %v", inputsSchemaMap)
+				inputTypesMap, err := getInputTypesMapFromCatalotItemSchema(inputsSchemaMap)
+				if err != nil {
+					return err
+				}
+
+				log.Printf("InputTypesMap: %v", inputTypesMap)
+				inputs, err = getInputsByType(v.(map[string]interface{}), inputTypesMap)
+				if err != nil {
+					return err
+				}
+			}
+		} else if blueprintId != "" {
+			blueprintVersion := ""
+			if v, ok := d.GetOk("blueprint_version"); ok {
+				blueprintVersion = v.(string)
+			}
+
+			if v, ok := d.GetOk("inputs"); ok {
+				inputsSchemaMap, err := getBlueprintSchema(apiClient, blueprintId, blueprintVersion)
+				if err != nil {
+					return err
+				}
+
+				log.Printf("Inputs Schema: %v", inputsSchemaMap)
+				inputTypesMap, err := getInputTypesMapFromBlueprintInputsSchema(inputsSchemaMap)
+				if err != nil {
+					return err
+				}
+
+				log.Printf("InputTypesMap: %v", inputTypesMap)
+				inputs, err = getInputsByType(v.(map[string]interface{}), inputTypesMap)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		reason := "Updated deployment inputs from vRA provider for Terraform."
+		err := runAction(d, apiClient, deploymentUuid, actionId, inputs, reason)
+		if err != nil {
+			return err
+		}
+
+		d.SetPartial("inputs")
+		log.Printf("Finished updating vra_deployment %s with inputs", name)
+	}
+
+	return nil
+}
+
+func getInputsByType(inputs map[string]interface{}, inputTypesMap map[string]string) (map[string]interface{}, error) {
+	log.Printf("Converting the input values to their types")
+	inputsByType := make(map[string]interface{})
+	var err error
+	for k, v := range inputs {
+		if t, ok := inputTypesMap[k]; ok {
+			log.Printf("input_key: %s, type: %#v", k, t)
+			switch strings.ToLower(t) {
+			case "boolean":
+				inputsByType[k], err = strconv.ParseBool(v.(string))
+			case "integer":
+				inputsByType[k], err = strconv.Atoi(v.(string))
+			default:
+				inputsByType[k] = v
+			}
+			if err != nil {
+				return nil, fmt.Errorf("cannot convert %v into %s", v, t)
+			}
+		} else {
+			inputsByType[k] = v
+		}
+	}
+	return inputsByType, nil
+}
+
+// Returns a map of string, string with input variables and their types defined in the vRA blueprints
+func getInputTypesMapFromCatalotItemSchema(schema map[string]interface{}) (map[string]string, error) {
```

Comment:
typo: getInputTypesMapFromCatalotItemSchema should be getInputTypesMapFromCatalogItemSchema

Reply:
Done.